server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1412,6 +1418,17 @@ private boolean checkForHypervisorIdUpdate(Consumer existing, ConsumerDTO incomi
         return false;
     }
 
+    private boolean isUsedByOwner(HypervisorIdDTO incomingId, Owner owner) {
+        if (incomingId == null) {
+            return false;
+        }
+        return owner.getConsumers().stream()
```

Comment:
What is the performance implication of this when an organization has 100k+ consumers?

Reply:
And why would you want to stream over all consumers vs getting a particular consumer with a more specific query?

Reply:
You are right. I didn't realize how many consumers there can be. I updated the query.